stable/app/templates/cron.yaml

```
@@ -1,12 +1,15 @@
 {{- if .Values.cron.enabled }}
-{{- range $job := .Values.cron.jobs }}
+{{- $fullName := include "app.fullname" . -}}
+{{- $appLabels := include "app.labels" . -}}
+{{- $appSelectorLabels := include "app.selectorLabels" . -}}
+{{- range $i, $job := .Values.cron.jobs }}
 apiVersion: batch/v1
 kind: CronJob
 metadata:
-  name: "{{ include "app.fullname" . }}-{{ $job.name }}"
-  namespace: {{ .Release.Namespace }}
+  name: {{ printf "%s-%d-%s" $.Release.Namespace $i $job.name | trunc 63 | trimSuffix "-" }}
```

```suggestion
  name: "{{ include "app.fullname" . | trunc 29 }}-{{ $job.name | trunc 29}}-{{ $i }}"
```

Limitation - only works for jobs till `999` (should be enough 🤞)

---

Can we have the index before the job name...

---

Sure, I am ok with both.

---

Updated; with index in the middle.